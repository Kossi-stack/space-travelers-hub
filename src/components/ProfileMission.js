import PropTypes from 'prop-types';

const ProfileMission = ({ mission }) => (
  <li key={mission.mission_id} className="profile__li">
    {mission.mission_name}
  </li>
);

ProfileMission.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.string.isRequired,
    mission_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }).isRequired,
};

export default ProfileMission;
