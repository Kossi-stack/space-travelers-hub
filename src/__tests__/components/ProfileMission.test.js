import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileMission from '../../components/ProfileMission';

it('renders Profile Mission Component correctly', () => {
  const mission = {
    mission_id: '9D1B7E0',
    mission_name: 'Thiacom',
    description: 'some description',
    reserved: false,
  };
  const tree = renderer
    .create(
      <Router>
        <ProfileMission key={mission.mission_id} mission={mission} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
