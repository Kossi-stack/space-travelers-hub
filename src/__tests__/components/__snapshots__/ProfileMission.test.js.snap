// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile Mission Component correctly 1`] = `
<li
  className="profile__li"
>
  Thiacom
</li>
`;
